### PR TITLE
fix: Persist Orchestrator Model settings in appsettings

### LINKED ISSUE #7946: Orchestrator:  Persist model settings data when Orchestrator is used
## Describe the bug
We need to persist the Orchestrator model data in Composer.

Right now, we always use the recommended(default) Orchestrator language models, but the defaults and behavior can change - we don't want to have users program to a model, and suddenly change the model when the default recommended model gets updated.

### PR DESCRIPTION
**Pull request by taicchoumsft**
## Description
Persist the model name in `appsettings.json`, which would look like this:
```
  "orchestrator": {
    "model": {
      "en_intent": "pretrained.20210521.microsoft.dte.01.06.int.en.onnx",
      "multilingual_intent": "pretrained.20210608.microsoft.dte.01.06.int.unicoder_multilingual.onnx"
    }
  },
```
These settings are only written when orchestrator is used for the first time.  The default model names  will be picked by checking against the Orchestrator models API.  The user can later alter these models by changing the model name in appsettings and the bot will use the new model.  If an invalid model is chosen, the bot will fail to start with a warning that the model is invalid. 

We only persist the model names in appsettings.  The `generated/orchestrator.settings.json` and orchestrator recognizer dialog file logic are the same.

## Task Item
closes #7946

### CODE REVIEW

File: Composer/packages/types/src/orchestrator.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+export type ModelTypes = 'en_intent' | 'multilingual_intent';
+
 export type OrchestratorModelRequest = {
-  kind: 'en_intent' | 'multilingual_intent';
+  kind: ModelTypes;
```
Comment (srinaath) [inline Composer/packages/types/src/orchestrator.ts:7]: Singular `ModelType`

**Review by srinaath — CHANGES_REQUESTED** [review]

File: Composer/packages/types/src/orchestrator.ts [hunk 2]
```diff
@@ -11,3 +13,17 @@ export enum DownloadState {
   ALREADYDOWNLOADED = 'ALREADYDOWNLOADED',
   DOWNLOADING = 'DOWNLOADING',
 }
+
+export interface IOrchestratorNLRList {
```
[the remark this answers is not in the document]
Comment (taicchoumsft) [reply Composer/packages/types/src/orchestrator.ts]: thanks, fixed!

File: Composer/packages/types/src/orchestrator.ts (hunk 1 above)
Comment (taicchoumsft) [reply Composer/packages/types/src/orchestrator.ts]: fixed!

File: Composer/packages/client/src/recoilModel/dispatchers/orchestrator.ts [hunk 3]
```diff
@@ -49,11 +56,15 @@ export const downloadModel = async (
   return await pollUntilDone(async () => (await httpClient.get(resp.data)).data !== DownloadState.DOWNLOADING, 5000);
 };
 
-export const availableLanguageModels = (recognizerFiles: RecognizerFile[], botSettings?: DialogSetting) => {
+export const availableLanguageModels = async (
```
Comment (hatpick) [inline Composer/packages/client/src/recoilModel/dispatchers/orchestrator.ts:59]: can you rename this to ```getAvailableLanguageModels```

Comment (taicchoumsft) [reply Composer/packages/client/src/recoilModel/dispatchers/orchestrator.ts]: thanks @hatpick , renamed.

**Review by hatpick — APPROVED** [review]

**Pull request merged**